src/sage/functions/special.py

```diff
@@ -216,11 +216,16 @@ class SphericalHarmonic(BuiltinFunction):
 
         sage: spherical_harmonic(1, 1, pi/2, pi).n()  # abs tol 1e-14                   # needs sage.symbolic
         0.345494149471335
-        sage: from scipy.special import sph_harm  # NB: arguments x and y are swapped   # needs scipy
         sage: import numpy as np                                                        # needs scipy
         sage: if int(np.version.short_version[0]) > 1:                                  # needs scipy
         ....:     np.set_printoptions(legacy="1.25")                                    # needs scipy
-        sage: sph_harm(1, 1, pi.n(), (pi/2).n())  # abs tol 1e-14                       # needs scipy sage.symbolic
+        sage: import scipy.version
+        sage: if scipy.version.version < '1.15.0':
+        ....:     from scipy.special import sph_harm # NB: arguments x and y are swapped   # needs scipy
+        ....:     sph_harm(1, 1, pi.n(), (pi/2).n())  # abs tol 1e-14                   # needs scipy sage.symbolic
+        ....: else:
+        ....:     from scipy.special import sph_harm_y                                  # needs scipy
+        ....:     sph_harm_y(1, 1, (pi/2).n(), pi.n()).item()  # abs tol 1e-9           # needs scipy sage.symbolic
```

Comment:
Instead of importing `scipy.version`, what about using a very pythonic way of doing thing like 
```
try:
   from scipy.special import sph_harm_y
   sph_harm_y(1, 1, (pi/2).n(), pi.n()).item()  # abs tol 1e-9
except ImportError:
  from scipy.special import sph_harm
  sph_harm(1, 1, pi.n(), (pi/2).n())  # abs tol 1e-14
```
?

Reply:
not sure, IMO this is easier to keep track of for future cleanups (when sage requires scipy>=1.15)

Reply:
Sounds like a fairly good argument.

Reply:
I agree...

It would be useful to have tags for different program versions to avoid having to use conditionals in the doctests (just like "32-bit" and "64-bit" labels). This is not only doctest but also documentation, and it looks kind of ugly. With labels, perhaps the manual could show only the version that matches currently installed software or something like that...